### PR TITLE
feat: use horust as the dev container supervisor

### LINKED ISSUE #930: feat: Better management of entrypoint process
## Description

Currently multiple processes are managed under tini by concatenating multiple commands. 

Cons:
- hard to inspect per-process logs
- hard to inspect errors when start the container

Candidate: https://github.com/F1bonacc1/process-compose 

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

* fix #930

### CODE REVIEW

File: pkg/lang/ir/supervisor.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/moby/buildkit/client/llb"
+
+	"github.com/tensorchord/envd/pkg/config"
+	"github.com/tensorchord/envd/pkg/types"
+)
+
+const (
+	horustTemplate = `
+name = "%[1]s"
+command = "%[2]s"
```
Comment (gaocegege) [inline pkg/lang/ir/supervisor.go:31]: We may need to set:

```
working-directory = "${WORKDIR}"
```

Comment (kemingy) [reply pkg/lang/ir/supervisor.go:31]: Fixed. Thanks.

**Comment by gaocegege** [comment]
Hi, is it ready to review?

**Comment by kemingy** [comment]
> Hi, is it ready to review?

yes, PTAL

File: base-images/horust.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM rust:bullseye as builder
+
+RUN git clone https://github.com/FedericoPonzi/Horust.git /app --depth=1
```
[the remark this answers is not in the document]
Comment (kemingy) [reply base-images/horust.Dockerfile:3]: Good idea. But I don't have permission. Can you help me?

[the remark this answers is not in the document]
Comment (gaocegege) [reply base-images/horust.Dockerfile:3]: Sure https://github.com/tensorchord/Horust

[the remark this answers is not in the document]
Comment (gaocegege) [reply base-images/horust.Dockerfile:3]: You can use the same secret as this repo to push to docker hub.

File: pkg/lang/ir/editor.go [hunk 3]
```diff
@@ -99,6 +104,11 @@ func (g Graph) generateRStudioCommand(workingDir string) []string {
 		return nil
 	}
 
+	// get from env if not set
+	// if len(workingDir) == 0 {
```
[the remark this answers is not in the document]
Comment (gaocegege) [reply pkg/lang/ir/editor.go:108]: We already have the same logic there https://github.com/tensorchord/envd/pull/1051/files#diff-2f089d619e15961bd13e9ad3bb23d7e716e24e0954f493a437dd8308c965c98cR84

Do we need the comments here?

[the remark this answers is not in the document]
Comment (kemingy) [reply pkg/lang/ir/editor.go:108]: That one is for jupyter. This is RStudio.

[the remark this answers is not in the document]
Comment (gaocegege) [reply pkg/lang/ir/editor.go:108]: Gotcha, LGTM

**Comment by gaocegege** [comment]
/lgtm

Thanks for your contribution! :tada: :+1:

**Pull request merged**